agent/agent.go

```diff
@@ -96,13 +100,27 @@ func NewAgent(config *Config) (*Agent, error) {
 	}
 	agent.allowList = allowList
 
+	agent.header = agent.buildTransportHeaderMap()
+	userCredentials := fmt.Sprintf("%s:%s", agent.config.User, agent.config.Password)
+	userCredentials = base64.StdEncoding.EncodeToString([]byte(userCredentials))
+	agent.header.Set("Authorization", "Basic "+userCredentials)
+	if agent.config.ProtobufSerialization {
+		pm := runtime.ProtoMarshaller{}
+		agent.header.Set("Content-Type", pm.ContentType())
+		agent.unmarshal = pm.Unmarshal
+		agent.marshal = pm.Marshal
+	} else {
+		agent.unmarshal = json.Unmarshal
+		agent.marshal = json.Marshal
+	}
+
 	return agent, nil
 }
 
 func (a *Agent) sendMessage(msg *transport.Message) {
 	logger.WithFields(logrus.Fields{
-		"type":    msg.Type,
-		"payload": string(msg.Payload),
```

Comment:
We cannot assume that this payload is always a check. Events, checks, and statsd metrics all call this function. We could however, pass along some identifier to the function, ex. `sendMessage(name string, type string, msg *transport.Message)`, which would allow us to log the name and type of whatever serialized payload is being sent.

As for logging the entire message at the debug level, I can definitely see how that would still be useful. However, I'm struggling to figure out a clean way to determine if the payload (at this point of the code) was serialized by protobuf or JSON. The string message of a protobuf payload would look something like this: 
```
{"component":"agentd","level":"debug","message":"\nN\n\techo hook(\nH\u0001Z\u0004miscr\u0014\n\u000fhook-with-asset\u0012\u00010\ufffd\u0001 \n\u0015check-with-hook-asset\u0012\u0007default\u001a\u003c\n\u001a\n\u000fhook-with-asset\u0012\u0007default\u0012\recho-asset.sh\u0018\u003c*\rstaging-asset پ\ufffd\ufffd\u0005","msg":"session - sending message","time":"2019-07-17T10:48:09-07:00"}
```
Applying `fmt.Sprintf("%x", protoMsg)` to the protobuf message to print its hex could be useful, do you have any ideas for how to differentiate a protobuf payload vs a JSON one?

Reply:
💩 	https://github.com/sensu/sensu-go/blob/a821698fe8dfda2b58e70e5c7d0b3b4d0f5c2bd1/backend/store/etcd/store.go#L214

Reply:
Hahaha @echlebek mentioned this hack when we were pairing! The alternate solution is to pass the content type with the message. I mean, this does work though!